tools/msms_extractor/msms_extractor.xml

```diff
@@ -0,0 +1,84 @@
+
+<tool id="msms_extractor" name="msms_extractor" version="1.0.0">
+  <description>Extract MS/MS scans from the mzML file(s) based on PSM report.</description>
+  <requirements>
+      <requirement type="package" version="3_0_9992">proteowizard</requirement>
+      <requirement type="package" version="3.4">pyteomics</requirement>
+  </requirements>
+  <command detect_errors="exit_code"><![CDATA[python msms_extractor.py $spectrumfile $psmreportfile $output ${spectrumfile.name.rsplit('.',1)[0]} $removeretain.doremoveretain $removeretain.num_random_scans]]></command>
+  <inputs>
+    <param name="spectrumfile" type="data" format="mzml">
+      <label>Input mzML File</label>
+    </param>
+    <param name="psmreportfile" type="data" format="tabular">
+      <label>Input PSM Report File</label>
+    </param>
+    
+    <conditional name="removeretain">
+        <param name="doremoveretain" type="select" display="radio" label="Retain or Remove the MS/MS scans that were identified in PSM report">
+            <option value="remove" selected="True">Remove the scans reported in the PSM Report File</option>
+            <option value="retain">Retain the scans reported in the PSM Report File </option>
+        </param>
+        
+        <when value="remove">
+            <param name="num_random_scans" type="hidden" value="0" />
+        </when>
+        <when value="retain">
+            <param name="num_random_scans" type="integer" label="Add N random scans in addition to those in the list" value="0" optional="false" />
```

Comment:
if you only have one param in a conditional you can make this param optional="true"

Reply:
Thanks @bgruening ! I have made all the changes as suggested. Thanks a lot again!